pkg/ruler/manager.go

```diff
@@ -106,24 +105,8 @@ func (r *DefaultMultiTenantManager) SyncRuleGroups(ctx context.Context, ruleGrou
 		RemoveFederatedRuleGroups(ruleGroups)
 	}
 
-	// Sync the rules to disk and then update the user's Prometheus Rules Manager.
-	// Since users are different, we can sync rules in parallel.
-	users := make([]string, 0, len(ruleGroups))
-	for userID := range ruleGroups {
-		users = append(users, userID)
-	}
-
-	// concurrenty.ForEachJob is a helper function that runs a function for each job in parallel.
-	// It cancel context of jobFunc once iteration is done.
-	// That is why the context passed to syncRulesToManager should be the global context not the context of jobFunc.
-	err := concurrency.ForEachJob(ctx, len(users), 10, func(_ context.Context, idx int) error {
-		userID := users[idx]
-		r.syncRulesToManager(ctx, userID, ruleGroups[userID])
-		return nil
-	})
-	if err != nil {
-		// The only error we could get here is a context canceled.
-		return
+	for userID, ruleGroup := range ruleGroups {
```

Comment:
I don't think this is correct. This was recently introduced.

Reply:
Thanks for catching it. Some mess with the merge. Re-synced the branches now and this diff is gone.